types/tx_msg.go

```diff
@@ -25,6 +25,16 @@ type (
 		GetSigners() []AccAddress
 	}
 
+	// Extension interface for Msg that require Amino Signing
+	// Replaces legacytx.LegacyMsg.
+	// Deprecated: use Msg instead.
+	HasAminoSigningCapability interface {
```

Comment:
I would propose to simply remove the 2 methods from LegacyMsg directly. It's slightly less API breaking, and it's a shorter and more used name that `HasAminoSigningCapability`